instrumentation/netty/netty-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/netty/v4_0/ChannelPipelineTest.java

```diff
@@ -0,0 +1,154 @@
+  // and https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4040
+  @ParameterizedTest
+  @CsvSource({"by instance", "by class", "by name", "first"})
+  void testRemoveOurHandler(String testName) throws Exception {
+    EmbeddedChannel channel = new EmbeddedChannel(new NoopChannelHandler());
+    ChannelPipeline channelPipeline = channel.pipeline();
+    HttpClientCodec handler = new HttpClientCodec();
+
+    // remove the default head and tail handlers
+    if (channelPipeline.first() != null) {
+      channelPipeline.removeFirst();
+    }
+    if (channelPipeline.last() != null) {
+      channelPipeline.removeLast();
+    }
+
+    // no handlers initially
+    assertEquals(0, channelPipeline.toMap().size());
+
+    // add handler
+    channelPipeline.addLast("http", handler);
+    assertEquals(handler, channelPipeline.first());
+    // our handler was also added
+    assertEquals("HttpClientTracingHandler", channelPipeline.last().getClass().getSimpleName());
+
+    if ("by instance".equals(testName)) {
+      channelPipeline.remove(handler);
+    } else if ("by class".equals(testName)) {
+      channelPipeline.remove(handler.getClass());
+    } else if ("by name".equals(testName)) {
+      channelPipeline.remove("http");
+    } else if ("first".equals(testName)) {
+      channelPipeline.removeFirst();
+    }
+
+    // removing handler also removes our handler
+    assertEquals(0, channelPipeline.toMap().size());
+  }
+
+  // regression test for
+  // https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/4040
+  @ParameterizedTest
+  @CsvSource({"by instance", "by class", "by name"})
+  void shouldReplaceHandler(String desc) {
+    EmbeddedChannel channel = new EmbeddedChannel(new NoopChannelHandler());
+    ChannelPipeline channelPipeline = channel.pipeline();
+    HttpClientCodec httpHandler = new HttpClientCodec();
+
+    // remove the default head and tail handlers
+    if (channelPipeline.first() != null) {
+      channelPipeline.removeFirst();
+    }
+    if (channelPipeline.last() != null) {
+      channelPipeline.removeLast();
+    }
+
+    // no handlers initially
+    assertEquals(0, channelPipeline.toMap().size());
+
+    NoopChannelHandler noopHandler = new NoopChannelHandler();
+    channelPipeline.addFirst("test", noopHandler);
+
+    // only the noop handler
+    assertEquals(1, channelPipeline.toMap().size());
+    assertEquals(noopHandler, channelPipeline.first());
+
+    if ("by instance".equals(desc)) {
+      channelPipeline.replace(noopHandler, "http", httpHandler);
+    } else if ("by class".equals(desc)) {
+      channelPipeline.replace(noopHandler.getClass(), "http", httpHandler);
+    } else if ("by name".equals(desc)) {
+      channelPipeline.replace("test", "http", httpHandler);
+    }
+
+    // noop handler was removed; http and instrumentation handlers were added
+    assertEquals(2, channelPipeline.toMap().size());
```

Comment:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/10410 removes `HttpClientTracingHandler` from the map so the size here will be `1` after you rebase your pr